utils/oc.py

```diff
@@ -213,18 +214,64 @@ def recycle_pods(self, dry_run, namespace, dep_kind, dep_resource):
         pods = self.get(namespace, 'Pods')['items']
 
         if dep_kind == 'Secret':
-            pods_to_recycle = [pod['metadata']['name'] for pod in pods
+            pods_to_recycle = [pod for pod in pods
                                if self.secret_used_in_pod(dep_name, pod)]
         else:
             raise RecyclePodsUnsupportedKindError(dep_kind)
 
+        recyclables = {}
         for pod in pods_to_recycle:
-            logging.info(['recycle_pod', namespace, pod])
-            if not dry_run:
-                self.delete(namespace, 'Pod', pod)
-                logging.info(['validating_pods', namespace])
-                self.validate_pods_ready(
-                    namespace, self.secret_used_in_pod, dep_name)
+            if pod['status']['phase'] == 'Succeeded':
+                # don't try to recycle a pod that completed successfully
+                continue
+            owner = self.get_obj_root_owner(namespace, pod)
+            kind = owner['kind']
+            if kind == 'ReplicationController':
```

Comment:
are ReplicationControllers still being used?

Reply:
I think I have found one or two.  We can't really control what tenants use, so imo better to handle it than not.

Reply:
but we do control it! `managedResourcesTypes` or `saasherder_object_blacklist` takes care of that.

Reply:
I'll remove the RC case.  Makes the code simplier.